src/typechecker.erl

```diff
@@ -153,6 +153,7 @@ compat_ty({type, _, any, []}, _, A, _TEnv) ->
     ret(A);
 compat_ty(_, {type, _, any ,[]}, A, _TEnv) ->
     ret(A);
+%% The clauses for term should never happen since term() is an alias of any().
 compat_ty({type, _, term, []}, _, A, _TEnv) ->
     ret(A);
 compat_ty(_, {type, _, term ,[]}, A, _TEnv) ->
```

Comment:
Now that term() is normalized to any(), these clauses should be deleted. We only need to handle types on head normal form.

Reply:
We don't want to keep these as safety? Then I'll remove them. At all places where I commented?

Reply:
Yes, remove all.

It it doesn't work, it means something is not normalized where it should be.